lightning/src/routing/gossip.rs

```diff
@@ -1182,11 +1185,16 @@ impl Readable for NodeAnnouncementInfo {
 			(4, rgb, required),
 			(6, alias, required),
 			(8, announcement_message, option),
-			(10, _addresses, optional_vec), // deprecated, not used anymore
+			(10, addresses, optional_vec),
 		});
-		let _: Option<Vec<SocketAddress>> = _addresses;
-		Ok(Self { features: features.0.unwrap(), last_update: last_update.0.unwrap(), rgb: rgb.0.unwrap(),
-			alias: alias.0.unwrap(), announcement_message })
+		Ok(Self {
+			features: features.0.unwrap(),
+			last_update: last_update.0.unwrap(),
+			rgb: rgb.0.unwrap(),
+			alias: alias.0.unwrap(),
+			addresses: addresses.unwrap_or(Vec::new()),
```

Comment:
Yeah, to avoid this I'd prefer one of the enum approaches that you mention in the PR description.

Reply:
We'd probably need three variants for:
- `NodeAnnouncement`
- `NodeAnnouncement` that isn't relayed (i.e., when `should_relay` is false) using a struct containing everything in `NodeAnnouncementInfo` but `announcement_message`
- `UnsignedNodeAnnouncement` (maybe with `excess_address_data` and `excess_data` cleared)

In that sense, `NodeAnnouncementInfo` would become an enum, though I'm not sure if there's anything that would make the undesirable. We'd also need accessor methods for the fields.

Reply:
That could work, though we'd need to clear `excess_address_data` and `excess_data` when not relaying, too. It would require that we copy the data to produce a `NodeAnnouncement` during relay, though it looks like the `RoutingMessageHandler` trait requires us to do that anyway, as currently designed.